install.sh

```diff
@@ -11,13 +11,47 @@ then
   abort "Bash is required to interpret this script."
 fi
 
+# Allow literal values for CI
+if [[ "${CI-}" =~ (yes|Yes|YES|true|True|TRUE|1) ]]
+then
+  CI=1
+elif [[ "${CI-}" =~ (no|No|NO|false|False|FALSE|0) ]]
+then
+  unset CI
+fi
+
+# Allow literal values for NONINTERACTIVE
+if [[ "${NONINTERACTIVE-}" =~ (yes|Yes|YES|true|True|TRUE|1) ]]
+then
+  NONINTERACTIVE=1
+elif [[ "${NONINTERACTIVE-}" =~ (no|No|NO|false|False|FALSE|0) ]]
+then
+  unset NONINTERACTIVE
+fi
+
+# Allow literal values for HAVE_SUDO_ACCESS
+if [[ "${HAVE_SUDO_ACCESS-}" =~ (no|No|NO|false|False|FALSE|1) ]]
+then
+  HAVE_SUDO_ACCESS=1 # this is the return value of a function (1 for fail, 0 for success)
+fi
+if [[ "${HAVE_SUDO_ACCESS-}" != "1" ]]
+then
+  unset HAVE_SUDO_ACCESS # we will check sudo permissions later in function have_sudo_access()
+fi
```

Comment:
There's a lot of repetition here. Can we extract the logic into a function we can just call a few times?

Reply:
I have refactored this into a function. It's a good idea to implement this in a separated function if we will introduce more environment variables in the future. By far, I think the previous version could be more readable (and fewer lines of code).

Reply:
We could just define two much simpler functions `is_true` and `is_false`, rather than the much more general `parse_literal`. This will lead to much less code but also limit the number of repetitions needed.